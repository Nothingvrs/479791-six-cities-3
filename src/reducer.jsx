import offers from './mocks/offers';

export const ActionType = {
  CHANGE_CITY: `change-city`,
  GET_OFFER: `get-offer`,
  CLEAR_OFFER: `clear-offer`
};

export const getOffersByCity = (initialOffers, cityName) =>
  initialOffers.filter((offer) => offer.city === cityName);

export const getCities = (initialOffers) => {
  const cities = new Set();
  initialOffers.forEach((offer) => cities.add(offer.city));
  return [...cities];
};

export const getOfferById = (initialOffers, id) => {
  const offer = initialOffers.find((newOffer) => newOffer.id === Number(id));
  if (!offer) {
    return null;
  }
  const newOffer = Object.assign({}, offer);
  newOffer.nearOffers = offer.nearOffers.map((offerId) =>
    initialOffers.find((offerItem) => offerItem.id === offerId)
  );
  return newOffer;
};

const initialState = {
  city: getCities(offers)[0],
  offers: getOffersByCity(offers, getCities(offers)[0]),
  citiesNames: getCities(offers)
};

export const ActionCreator = {
  changeCity: (cityName) => {
    const newOffers = getOffersByCity(offers, cityName);

    return {type: ActionType.CHANGE_CITY, payload: {offers: newOffers, city: cityName}};
  },
  getCardDetails(id) {

    const newOffer = getOfferById(offers, id);

    return {type: ActionType.GET_OFFER, payload: newOffer};
  },
  clearOffer() {
    return {type: ActionType.CLEAR_OFFER, payload: ``};
  }
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return Object.assign({}, state, {city: action.payload.city, offers: action.payload.offers});
    case ActionType.GET_OFFER:
      return Object.assign({}, state, {offer: action.payload});
    case ActionType.CLEAR_OFFER:
      return Object.assign({}, state, {offer: action.payload});
  }

  return state;
};
