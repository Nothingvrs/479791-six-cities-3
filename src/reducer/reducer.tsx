import {combineReducers} from "redux";
import {userReducer as user} from './user/user-reducer';
import {dataReducer as data} from "./data/data-reducer";
const rootReducer = combineReducers({
  user,
  data
});

export default rootReducer;
