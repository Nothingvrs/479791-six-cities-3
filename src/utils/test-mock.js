export const mockTestData = {
  numberOfOffers: 32,
  cards: [
    `Canal View Prinsengracht`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`
  ]
};

export const mockCards = [
  {
    id: 0,
    name: `Wood and stone place`,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 35,
    isInBookmark: false,
    mark: 2,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,

    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 5,
    avgMark: 5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location `,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 555,
    isInBookmark: true,
    mark: 3,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3809553943508, 4.939309666406198]
  }
];


export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};


