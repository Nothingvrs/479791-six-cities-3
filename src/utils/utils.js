import PropTypes from "prop-types";

export const cardPropTypes = PropTypes.shape({
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf([`Apartment`, `Private room`]).isRequired,
  imgs: PropTypes.arrayOf(PropTypes.string).isRequired,
  price: PropTypes.number.isRequired,
  isInBookmark: PropTypes.bool.isRequired,
  mark: PropTypes.oneOf([...new Array(6)].map((_, i) => i)).isRequired,
  isPremium: PropTypes.bool.isRequired,
  bedroomNo: PropTypes.number.isRequired,
  capacity: PropTypes.number.isRequired,
  facilities: PropTypes.arrayOf(PropTypes.string).isRequired,
  descriptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  reviews: PropTypes.number.isRequired,
  avgMark: PropTypes.number.isRequired,
  hostUser: PropTypes.shape({
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired
  }).isRequired}).isRequired;
