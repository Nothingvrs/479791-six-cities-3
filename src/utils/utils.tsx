export interface CommentModel {
  id: number;
  comment: string;
  mark: number;
  date: string;
  user: UserModel;
}

interface HostUserModel {
  name: string;
  img: string;
  isPro: boolean;
}

interface LocationModel {
  latitude: number;
  longitude: number;
  zoom: number;
}

export interface UserModel {
  id: number;
  email: string;
  name: string;
  img: string;
  isPro: boolean;
}

export interface CityModel {
  name?: string;
  location: LocationModel;
}

export interface CardModel {
  id: number;
  name: string;
  type: string;
  previewImg: string;
  imgs: string [];
  price: number;
  isInBookmark: boolean;
  mark: number;
  isPremium: boolean;
  bedroomNo: number;
  capacity: number;
  facilities: string [];
  descriptions: string [];
  reviews: number;
  avgMark: number;
  comments: CommentModel [];
  hostUser: HostUserModel;
  city: CityModel;
  addressCoords: [number, number];
  nearOffers?: CardModel [];
}


export const offerAdapter = (offer) => {
  return {
    id: offer.id,
    name: offer.title,
    type: offer.type.charAt(0).toUpperCase() + offer.type.slice(1),
    previewImg: offer.preview_image,
    imgs: offer.images,
    price: offer.price,
    isInBookmark: offer.is_favorite,
    mark: offer.rating,
    isPremium: offer.is_premium,
    bedroomNo: offer.bedrooms,
    capacity: offer.max_adults,
    facilities: offer.goods,
    descriptions: [offer.description],
    hostUser: {
      id: offer.host.id,
      name: offer.host.name,
      img: offer.host.avatar_url,
      isPro: offer.host.is_pro
    },
    addressCoords: [offer.location.latitude, offer.location.longitude],
    city: offer.city
  };
};

export const commentAdapter = (comment) => ({
  id: comment.id,
  comment: comment.comment,
  mark: comment.rating,
  date: comment.date,
  user: userAdapter(comment.user)
});

export const userAdapter = (user) => ({
  id: user.id,
  email: user.email,
  name: user.name,
  img: user.avatar_url,
  isPro: user.is_pro
});

export const getCities = (initialOffers) => {
  const citiesNames = new Set();
  const cities = [];
  initialOffers.forEach((offer) => {
    if (!citiesNames.has(offer.city.name)) {
      citiesNames.add(offer.city.name);
      cities.push(offer.city);
    }
  });
  return cities;
};
