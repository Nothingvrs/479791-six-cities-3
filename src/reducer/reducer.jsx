import {offerAdapter} from '../utils/utils';

export const getCities = (initialOffers) => {
  const citiesNames = new Set();
  const cities = [];
  initialOffers.forEach((offer) => {
    if (!citiesNames.has(offer.city.name)) {
      citiesNames.add(offer.city.name);
      cities.push(offer.city);
    }
  });
  return cities;
};

export const Action = {
  SET_HOVERED: `set-hovered`,
  RESET_HOVERED: `reset-hovered`,
  SET_CITY: `set-city`,
  SET_FILTER: `set-filter`,
  RESET_FILTER: `reset-filter`,
  GET_OFFERS: `get-offers`
};

export const ActionCreator = {
  setHovered(id) {
    return {type: Action.SET_HOVERED, payload: id};
  },
  resetHovered() {
    return {type: Action.RESET_HOVERED};
  },
  setFilter(filterName) {
    return {type: Action.SET_FILTER, payload: filterName};
  },
  resetFilter() {
    return {type: Action.RESET_FILTER};
  },
  setCity(cityName) {
    return {type: Action.SET_CITY, payload: cityName};
  },
  getOffersFromApi() {
    return (dispatch, state, api) => {
      api.get(`/hotels`).then((response) => {
        dispatch({
          type: Action.GET_OFFERS,
          payload: response.data.map((offer) => offerAdapter(offer))
        });
      });
    };
  }
};

const initialState = {
  filterName: `popular`,
  hoveredId: -1
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.SET_HOVERED:
      return Object.assign({}, state, {hoveredId: action.payload});
    case Action.RESET_HOVERED:
      return Object.assign({}, state, {hoveredId: -1});
    case Action.SET_FILTER:
      return Object.assign({}, state, {filterName: action.payload});
    case Action.RESET_FILTER:
      return Object.assign({}, state, {filterName: `popular`});
    case Action.SET_CITY:
      return Object.assign({}, state, {city: action.payload});
    case Action.GET_OFFERS:
      return Object.assign({}, state, {
        offers: action.payload,
        city: getCities(action.payload)[0],
        citiesNames: getCities(action.payload),
        isLoaded: true
      });
  }
  return state;
};
