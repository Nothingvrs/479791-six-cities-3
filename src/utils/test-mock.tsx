import {CardModel, getCities} from './utils';
import {Authorization} from '../reducer/user/user-reducer';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {createApi} from '../api';

export const userData = {
  id: 1,
  email: `laura@example.com`,
  name: `laura`,
  img: `/static/avatar/7.jpg`,
  isPro: false
};
export const mockCards: CardModel[] = [
  {
    id: 0,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 120,
    isInBookmark: true,
    mark: 4,
    isPremium: true,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3909553943508, 4.85309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 1,
    name: `Wood and stone place`,
    type: `Private room`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 80,
    isInBookmark: true,
    mark: 0,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`
    ],
    descriptions: [
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Natalia`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.369553943508, 4.85309666406198],
    comments: [
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-02.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cabel TV`
    ],
    descriptions: [
      `Chalets Alpins- 15 Chemin des Skieurs is located in Stoneham, 9.7 km from Les Marais du Nord, 9.7 km from Camping Stoneham, as well as 19.3 km from Nordique Spa Stoneham. The air-conditioned accommodation is an 8-minute walk from Halte O Spa, and guests benefit from private parking available on site and free WiFi.`,
      `Aventures Nord-Bec is 24.1 km from the chalet, while Jacques-Cartier National Park is 48.3 km from the property. The nearest airport is Qu√©bec City Jean Lesage International Airport, 40.2 km from Chalets Alpins- 15.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Olga`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3909553943508, 4.929309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-03.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cabel TV`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      `The chalet features 5 separate bedrooms, 1 bathroom, a fully equipped kitchen with a dining area and dishwasher, and a living room with a TV.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Natalia`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3809553943508, 4.939309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 1,
        date: `2020-03-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  }
];

export const mockCardWithServerFormat = {
  id: 3,
  title: `Nice, cozy, warm big bed apartment`,
  type: `Apartment`,
  preview_image: `/img/apartment-01.jpg`,
  images: [
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-02.jpg`,
    `/img/apartment-03.jpg`,
    `/img/studio-01.jpg`,
    `/img/room.jpg`
  ],
  price: 180,
  /* eslint @typescript-eslint/camelcase: ["error", {properties: "never"}] */
  is_favorite: false,
  rating: 5,
  is_premium: false,
  bedrooms: 3,
  max_adults: 4,
  goods: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`
  ],
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  reviews: 2,
  avgMark: 4.5,
  host: {
    id: 1,
    name: `Natalia`,
    avatar_url: `/img/avatar-angelina.jpg`,
    is_pro: true
  },
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  location: {
    latitude: 52.3809553943508,
    longitude: 4.939309666406198,
    zoom: 13
  }
};

export const mockCommentWithServerFormat = {
  comment:
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  rating: 4,
  user: {
    avatar_url: `img/1.png`,
    id: 4,
    is_pro: false,
    name: `Max`
  }
};

export const mockCommentAdapted = {
  comment:
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  mark: 4,
  user: {
    img: `img/1.png`,
    email: undefined,
    id: 4,
    isPro: false,
    name: `Max`
  }
};

export const adaptedMockData = {
  id: 3,
  name: `Nice, cozy, warm big bed apartment`,
  type: `Apartment`,
  previewImg: `/img/apartment-01.jpg`,
  imgs: [
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-02.jpg`,
    `/img/apartment-03.jpg`,
    `/img/studio-01.jpg`,
    `/img/room.jpg`
  ],
  price: 180,
  isInBookmark: false,
  mark: 5,
  isPremium: false,
  bedroomNo: 3,
  capacity: 4,
  facilities: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`
  ],
  descriptions: [
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  ],
  hostUser: {
    id: 1,
    name: `Natalia`,
    img: `/img/avatar-angelina.jpg`,
    isPro: true
  },
  addressCoords: [52.3809553943508, 4.939309666406198],
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  }
};

export const mockCities = [
  {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  }
];

export const logInMockData = {
  password: `123`,
  email: `1513@123.ru`
};

export const mockComment = {
  id: 1,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  mark: 4,
  date: `2020-01-26T13:51:50.417Z`,
  user: userData
};

export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const getTestStore = () => {
  const initialState = {
    data: {
      city: mockCities[0],
      offers: mockCards,
      citiesNames: getCities(mockCards),
      hoveredId: -1,
      filterName: `popular`
    },
    user: userData,
    authorizationStatus: Authorization.NO_AUTH
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();
  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
