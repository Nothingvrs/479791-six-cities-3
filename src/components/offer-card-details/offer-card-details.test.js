import OfferCardDetails from './offer-card-details.jsx';
import React from 'react';
import {mockCards} from '../../utils/test-mock';
import Enzyme, {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import EnzymeReactAdapter from 'enzyme-adapter-react-16';
import {getCities, getOffersByCity} from '../../reducer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';

Enzyme.configure({adapter: new EnzymeReactAdapter()});

const initialState = {
  city: getCities(mockCards)[0],
  offers: getOffersByCity(mockCards, getCities(mockCards)[0]),
  citiesNames: getCities(mockCards)
};

const reducer = (state = initialState) => {
  return state;
};
const store = createStore(reducer);

Enzyme.configure({adapter: new EnzymeReactAdapter()});

it(`OfferCardDetails successfully rendered`, () => {
  const mockHistory = {push: jest.fn};
  const mockMatch = {
    params: {
      id: 0
    }
  };
  const tree = mount(
      <Provider store={store}>
        <OfferCardDetails
          card={mockCards[0]}
          onHeaderClick={() => {}}
          history={mockHistory}
          match={mockMatch}
        />
      </Provider>
  );
  expect(toJson(tree, {mode: `deep`})).toMatchSnapshot();
});
