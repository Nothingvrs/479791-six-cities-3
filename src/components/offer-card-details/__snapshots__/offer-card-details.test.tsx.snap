// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCardDetails successfully rendered 1`] = `
<Connect(OfferCardDetail)
  history={
    Object {
      "push": [Function],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 0,
      },
    }
  }
>
  <div
    className="page"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <Link
              className="header__logo-link"
              to="/"
            >
              <LinkAnchor
                className="header__logo-link"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="header__logo-link"
                  href="/"
                  onClick={[Function]}
                >
                  <img
                    alt="6 cities logo"
                    className="header__logo"
                    height="41"
                    src="/img/logo.svg"
                    width="81"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <Link
                  className="header__nav-link header__nav-link--profile"
                  to="/login"
                >
                  <LinkAnchor
                    className="header__nav-link header__nav-link--profile"
                    href="/login"
                    navigate={[Function]}
                  >
                    <a
                      className="header__nav-link header__nav-link--profile"
                      href="/login"
                      onClick={[Function]}
                    >
                      <div
                        className="header__avatar-wrapper user__avatar-wrapper"
                      />
                      <span
                        className="header__login"
                      >
                        Sign in
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--property"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
              key="0"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/apartment-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="1"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/apartment-02.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="2"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/apartment-03.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="3"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/studio-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="4"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/room.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            <div
              className="property__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Beautiful & luxurious apartment at great location
              </h1>
              <Link
                to="/login"
              >
                <LinkAnchor
                  href="/login"
                  navigate={[Function]}
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    <button
                      className="property__bookmark-button button property__bookmark-button--active"
                      data-test="test-add-to-favorite"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="property__bookmark-icon"
                        height="33"
                        width="31"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In
                         bookmarks
                      </span>
                    </button>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                4
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                Apartment
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                1
                 Bedroom
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                1
                 adult
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                120
              </b>
              <span
                className="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                  key="0"
                >
                  Wi-Fi
                </li>
                <li
                  className="property__inside-item"
                  key="1"
                >
                  Heating
                </li>
                <li
                  className="property__inside-item"
                  key="2"
                >
                  Kitchen
                </li>
                <li
                  className="property__inside-item"
                  key="3"
                >
                  Fridge
                </li>
                <li
                  className="property__inside-item"
                  key="4"
                >
                  Washing machine
                </li>
                <li
                  className="property__inside-item"
                  key="5"
                >
                  Coffee machine
                </li>
                <li
                  className="property__inside-item"
                  key="6"
                >
                  Dishwasher
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="//img/avatar-angelina.jpg"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  Angelina
                </span>
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                  key="0"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                </p>
                <p
                  className="property__text"
                  key="1"
                >
                  An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="near-places__list places__list"
          />
        </section>
      </div>
    </main>
  </div>
</Connect(OfferCardDetail)>
`;
